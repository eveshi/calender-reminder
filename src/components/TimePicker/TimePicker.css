*,
*::after,
*::before {
  margin: 0;
  padding: 0;
  box-sizing: border-box; }

html {
  font-size: 62.5%; }
  @media only screen and (max-width: 400px) {
    html {
      font-size: 50%; } }

body {
  font-family: 'Roboto', sans-serif;
  background-color: #F5F5F4; }

a svg {
  height: 4rem;
  width: 4rem;
  text-decoration: none;
  fill: #5F513F; }
  a svg:hover {
    background-color: transparent; }

@font-face {
  font-family: 'Roboto', sans-serif;
  src: url("https://fonts.googleapis.com/css?family=Roboto"); }

.TimePicker {
  display: flex;
  flex-direction: row;
  align-items: center; }
  .TimePicker_select {
    font-size: 2.4rem;
    margin: auto .5rem;
    background-color: transparent;
    border: none;
    color: #5F513F; }
    .TimePicker_select:focus {
      outline: none; }
