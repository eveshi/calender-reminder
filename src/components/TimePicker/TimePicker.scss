@import '../../stylesheets/main.scss';

.TimePicker {
  display: flex;
  flex-direction: row;
  align-items: center;

  &_select {
    font-size: 2.4rem;
    margin: auto .5rem;
    background-color: transparent;
    border: none;
    color: $color-secondary-darker;

    &:focus {
      outline: none;
    }
  }
}