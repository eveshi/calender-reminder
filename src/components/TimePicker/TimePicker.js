import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import * as actions from '../../store/actions/index';

import HourPicker from './Pickers/HourPicker';
import MinutePicker from './Pickers/MinutePicker';

import './TimePicker.css';

export const TimePicker = (props) => {
  const {
    hourPicked,
    minutePicked,
    hourArray,
    minuteArray,
    changeSessionHour,
    changeSessionMinute,
  } = props;

  const timeChangeHandler = (event, selector) => {
    const { value } = event.target;

    if (selector === 'hour') {
      changeSessionHour(value);
    }

    if (selector === 'minute') {
      changeSessionMinute(value);
    }
  };

  return (
    <form className="TimePicker">
      <HourPicker
        hourPicked={hourPicked}
        hourArray={hourArray}
        onChange={event => timeChangeHandler(event, 'hour')}
      />
      <p className="TimePicker_select">:</p>
      <MinutePicker
        minutePicked={minutePicked}
        minuteArray={minuteArray}
        onChange={event => timeChangeHandler(event, 'minute')}
      />
    </form>
  );
};

TimePicker.propTypes = {
  hourPicked: PropTypes.string.isRequired,
  minutePicked: PropTypes.string.isRequired,
  hourArray: PropTypes.arrayOf(PropTypes.string).isRequired,
  minuteArray: PropTypes.arrayOf(PropTypes.string).isRequired,
  changeSessionHour: PropTypes.func.isRequired,
  changeSessionMinute: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  hourPicked: state.sessionState.hour,
  minutePicked: state.sessionState.minute,
  hourArray: state.sessionState.hourArray,
  minuteArray: state.sessionState.minuteArray,
});

const mapActionToProps = dispatch => ({
  changeSessionHour: hour => dispatch(actions.changeSessionHour(hour)),
  changeSessionMinute: minute => dispatch(actions.changeSessionMinute(minute)),
});


export default connect(mapStateToProps, mapActionToProps)(TimePicker);
