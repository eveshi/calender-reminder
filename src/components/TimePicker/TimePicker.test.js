import React from 'react';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { TimePicker } from './TimePicker';

configure({ adapter: new Adapter() });

const props = {
  hourPicked: '00',
  minutePicked: '00',
  hourArray: ['00', '01'],
  minuteArray: ['00', '01'],
  changeSessionHour: jest.fn(),
  changeSessionMinute: jest.fn(),
};

describe('snapshot', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<TimePicker {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
