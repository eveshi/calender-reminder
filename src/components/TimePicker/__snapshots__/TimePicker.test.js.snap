// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders correctly 1`] = `
<form
  className="TimePicker"
>
  <select
    className="TimePicker_select"
    name="hour"
    onChange={[Function]}
    value="00"
  >
    <option
      value="00"
    >
      00
    </option>
    <option
      value="01"
    >
      01
    </option>
  </select>
  <p
    className="TimePicker_select"
  >
    :
  </p>
  <select
    className="TimePicker_select"
    name="minute"
    onChange={[Function]}
    value="00"
  >
    <option
      value="00"
    >
      00
    </option>
    <option
      value="01"
    >
      01
    </option>
  </select>
</form>
`;
